packages/ubuntu_desktop_installer/lib/utils/version_extractor.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+class VersionExtractor {
+  static VersionExtractor? _instance;
+
+  static VersionExtractor get instance {
+    if (_instance == null) {
+      _instance = VersionExtractor._();
+    }
+    return _instance!;
+  }
+
+  VersionExtractor._();
+
+  String loadedVersion = '';
+
+  String _dummy() {
+    return 'Ubuntu';
+  }
+
+  String extractLocalVersion(String content) {
+    return content.substring(content.indexOf("\"") + 1, content.length - 1);
+  }
+
+  String getVersion({
+    String path = '/cdrom/.disk/info',
+    String localPath = '/etc/os-release',
+  }) {
+    if (loadedVersion.isNotEmpty) {
+      return loadedVersion;
+    }
+
+    final fileOnCdrom = File(path);
+    final localVersionFile = File(localPath);
```

Comment:
No need to instantiate that File object until after failing to extract the info from the ISO.